iridauploader/api/api_calls.py

```diff
@@ -430,13 +430,16 @@ def get_samples(self, project_id):
                 # use name and description from dictionary as base parameters when creating sample
                 sample_name = sample_dict['sampleName']
                 sample_desc = sample_dict['description']
+                sample_identifier = int(sample_dict['identifier'])
                 # remove them from the dict so we don't have useless duplicate data
                 del sample_dict['sampleName']
                 del sample_dict['description']
+                del sample_dict['identifier']
                 sample_list.append(model.Sample(
                     sample_name=sample_name,
                     description=sample_desc,
-                    samp_dict=sample_dict
+                    samp_dict=sample_dict,
+                    sample_number=sample_identifier
```

Comment:
Wouldn't it make sense to keep this as `sample_id` or `sample_identifier` for consistency. I see it is used in a method below called `get_sample_id`.

Reply:
Ok so this led me down a rabbit hole of code I had not seen in years. Turns out sample_number is actually a special value we use when parsing for certain sequencers that have their own ordering of samples. I had assumed that sample_number == sample_id but that was not the case.

I have made sample_id its own property on the Sample object and updated the documentation to better explain this.